piccolo/tablestorage.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+import typing as t
+
+from piccolo.apps.schema.commands.generate import get_output_schema
+from piccolo.table import Table, TableMetaclass
+
+
+class Immutable(object):
+    def _immutable(self, *arg, **kw) -> TypeError:
+        raise TypeError("%s object is immutable" % self.__class__.__name__)
+
+    __delitem__ = __setitem__ = __setattr__ = _immutable
+
+
+class ImmutableDict(Immutable, dict):
```

Comment:
This is cool, but do you think we need it to be immutable, vs a standard dictionary?

If it for the user's safety, in case they somehow mangled this virtual schema, and then inserted garbage data into the database?

Reply:
Yes, exactly. Overriding default data types are not a very good idea. But I couldn't find a better way to ensure data safety.
I'll gladly try alternative ways.

Reply:
I can't think of an alternative. Python has frozen sets, and frozen data classes, but not sure how to make them applicable here.